### ISSUE TITLE
Slots are not accurate

### ISSUE THREAD

**Issue by nickidw** [tinuva/ha-coct-loadshedding#4]
## Version of the custom_component

1.0.7

## Configuration

```yaml

Add your logs here.

```

## Describe the bug

According to the sensor my next loadshedding slot is only tomorrow 4 Oct at 08:00, but it's actually at 00:00, with nothing at 08:00

## Debug log

```text

Add your logs here.

```
![image](https://user-images.githubusercontent.com/12016514/193661228-a17a1932-5689-4d6a-9cfd-d3c9ad25693a.png)

**Comment by tinuva** [comment]
Hi @nickidw 

From looking at the screenshot, it looks like the upcoming loadshedding slots are based on the current stage, stage 2, where as the loadshedding slot you mention is for stage3.

I haven't had time to have the upcoming slots take into account the stage changes and that would be the reason for what you observe here. It is definitely on a todo list to fix in the future.

**Comment by nickidw** [comment]
ok cool, thanks. I'm busy with my own project to determine these, it uses
outputs from https://github.com/beyarkay/eskom-calendar.

Have not tested it yet, that's next on my list, but have proof of concept
code that processes the outputs with some todo left. Welcome to look here,
but dont judge :-D https://github.com/nickidw/ha-coct-loadshedding

**Comment by tinuva** [comment]
Well in honesty, I am using EskomSePush's API for private individuals.

See: https://github.com/tinuva/home-assistant-config/blob/master/packages/energy/eskomsepush.yaml

This give me the most accurate upcoming loadshedding slots:

![Screenshot 2022-10-04 at 14 03 05](https://user-images.githubusercontent.com/5417682/193814593-c42b4873-cebd-436a-afe9-d7c5f97fc273.png)

**Comment by nickidw** [comment]
Cool, thanks!

**Comment by Zodiac-69** [comment]
I have also seen discrepancies.
The Stage is correct, i am on CoCT, but the time-slot is not.
I compared yours with [dalehumby/Eskom-Loadshedding-NodeRED](https://github.com/dalehumby/Eskom-Loadshedding-NodeRED), and his time-slot calculation are spot-on.

Not sure if it is https://d42sspn7yra3u.cloudfront.net// that is not reporting the correct time-slot for CoCT.

**Comment by tinuva** [comment]
Yeah unfortunately sometimes City of Cape town doesnt update the endpoint. The same issue then happen in the CoCT loadshedding app.

**Comment by Zodiac-69** [comment]
Is there a way to notify the CoCT that their portal has errors? It would be fantastic to get hold of a person that is responsible for this and see if we can assist in automating the process as far as possible.
It is a pain to use multiple apps to get the correct schedule.

With more people using "automation" to automate due to load shedding, the quality of the data is getting more and more important.

**Comment by geeneo** [comment]
Seems we are suck, no updates since the 17th :(
